broker-daemon/relayer/market-watcher.js

```diff
@@ -64,6 +64,8 @@ class MarketWatcher extends EventEmitter {
   async handleResponse (response) {
     const { RESPONSE_TYPES } = this
 
+    // TODO: I don't think this will work well if there is more than one event queued since they all
```

Comment:
Can you explain `why` in more detail. Do you think this would cause us to get into a weird state? What would be the situation?